.github/workflows/ci.yml

```diff
@@ -16,21 +16,21 @@ on:
     - synchronize
 
 env:
-  GOLANGCI_LINT_VERSION: 'v1.54.2'
+  GOLANGCI_LINT_VERSION: 'v1.55.2'
   NFPM_VERSION: 'v2.32.0'
 
 jobs:
   lint:
     name: Lint
     runs-on: ubuntu-22.04
     steps:
-      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1 
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
the recommendation is to use a tag and not a commit sha or branch by github

https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-tags-for-release-management
https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-a-commits-sha-for-release-management

Reply:
for security reasons its recommended to use the full sha
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Reply:
will address in a separate PR